### ISSUE TITLE
Improve exception for object property that does not exist

### ISSUE THREAD

**Issue by BernieWhite** [microsoft/PSRule.Rules.Azure#362]
**Description of the issue**

When accessing an object property that does not exist within a template (i.e. provided by a parameter), an ambiguous error message is reported.

**Expected behaviour**

The exception message should be clearer that the property does not exist.

**Error output**

```text
Object reference not set to an instance of an object.
```

**Module in use and version:**

- Module: PSRule.Rules.Azure
- Version: **0.11.0**
